gadget/update.go

```diff
@@ -1109,11 +1123,22 @@ func MockVolumeStructureToLocationMap(f func(_ GadgetData, _ Model, _ map[string
 // use indirection to allow mocking
 var volumeStructureToLocationMap = volumeStructureToLocationMapImpl
 
-func volumeStructureToLocationMapImpl(old GadgetData, mod Model, laidOutVols map[string]*LaidOutVolume) (map[string]map[int]StructureLocation, error) {
+// volumeStructureToLocationMapImpl builds a map of gadget structures
+// to locations and to matched disk structures. For the locations, the
+// first key is the volume name, and the second key is the structure's
+// index in the list of structures on that volume. The value is the
+// StructureLocation that can actually be used to perform the
+// lookup/update in applyUpdates. For the matched disk, the first key
+// is the volume name and the second key is the yaml index of the
+// structure in the gadget definition. The value is the disk structure
```

Comment:
written this way is not very clear but looking at the code both indexes are the volume yaml index? both maps are indexed by [volName][volYamlIndex]?  I wonder whether it would make sense to have StructureLocation also carry (optionally?) the OnDiskStucture and have only one map? or is that silly? or merge instead StructureLocation and OnDiskStructure?

Reply:
or derive one from the other perhaps? instead of build them in parallel?